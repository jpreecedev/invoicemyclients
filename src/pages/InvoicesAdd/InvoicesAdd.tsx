import React from 'react'

import { AuthenticatedPage } from '../../components/AuthenticatedPage'
import { Card } from '../../components/Card'
import { Invoice } from '../../components/Invoice'

const InvoicesAddPage = () => {
  return (
    <AuthenticatedPage title="Create A New Invoice">
      <p>
        You are raising a new invoice for <strong>Jon Preece Development Ltd</strong>.
      </p>
      <Card>
        <Invoice onAction={console.log} />
      </Card>
    </AuthenticatedPage>
  )
}

export { InvoicesAddPage }
