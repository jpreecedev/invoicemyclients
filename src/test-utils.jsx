import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
