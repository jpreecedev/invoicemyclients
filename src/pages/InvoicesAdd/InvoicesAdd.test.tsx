import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { InvoicesAddPage } from './InvoicesAdd'

describe('<InvoicesAdd /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<InvoicesAddPage />).container).toMatchSnapshot()
  })
})
