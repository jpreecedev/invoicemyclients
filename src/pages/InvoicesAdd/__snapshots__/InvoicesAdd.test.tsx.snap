// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvoicesAdd /> tests should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        InvoiceMyClients.com
      </a>
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            class="fe fe-x"
          />
        </button>
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/clients"
            >
              Clients
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/invoices"
            >
              Invoices
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/nagging"
            >
              Nagging
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
        <a
          class="navbar-btn btn btn-sm btn-primary lift ml-auto"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <main
    class="mt-5 mb-10 pb-10"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h1>
            Create A New Invoice
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <p>
            You are raising a new invoice for 
            <strong>
              Jon Preece Development Ltd
            </strong>
            .
          </p>
          <div
            class="card shadow-light-lg card-border border-success mb-5"
          >
            <div
              class="card-body"
            >
              
              <form>
                <div
                  class="container"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-12 col-md-4 form-group"
                    >
                      <label
                        for="invoiceDate"
                      >
                        Invoice Date
                      </label>
                      <input
                        autocomplete="off"
                        class="form-control"
                        data-testid="invoiceDate"
                        id="invoiceDate"
                        name="invoiceDate"
                        placeholder="Invoice Date"
                        type="date"
                      />
                    </div>
                    <div
                      class="col-12 col-md-4 form-group"
                    >
                      <label
                        for="invoiceNumber"
                      >
                        Invoice Number #
                      </label>
                      <input
                        autocomplete="off"
                        class="form-control"
                        data-testid="invoiceNumber"
                        id="invoiceNumber"
                        name="invoiceNumber"
                        placeholder="Invoice Number #"
                        type="text"
                      />
                    </div>
                    <div
                      class="col-12 col-md-4 form-group"
                    >
                      <label
                        for="dueDate"
                      >
                        Due Date
                      </label>
                      <input
                        autocomplete="off"
                        class="form-control"
                        data-testid="dueDate"
                        id="dueDate"
                        name="dueDate"
                        placeholder="Due Date"
                        type="date"
                      />
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="col-12 col-md-4 form-group col-auto ml-auto"
                    >
                      <label
                        for="poNumber"
                      >
                        PO Number #
                      </label>
                      <input
                        autocomplete="off"
                        class="form-control"
                        data-testid="poNumber"
                        id="poNumber"
                        name="poNumber"
                        placeholder="PO Number #"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <table
                      class="table table-sm table-borderless table-responsive mt-5"
                    >
                      <thead
                        class="thead-light mb-2"
                      >
                        <tr>
                          <th
                            scope="col"
                          >
                            <span
                              class="text-uppercase p-2 text-black-50 text-nowrap"
                            >
                              Item
                            </span>
                          </th>
                          <th
                            scope="col"
                          >
                            <span
                              class="text-uppercase p-2 text-black-50 text-nowrap"
                            >
                              Description
                            </span>
                          </th>
                          <th
                            scope="col"
                          >
                            <span
                              class="text-uppercase p-2 text-black-50 text-nowrap"
                            >
                              Unit Cost
                            </span>
                          </th>
                          <th
                            scope="col"
                          >
                            <span
                              class="text-uppercase p-2 text-black-50 text-nowrap"
                            >
                              Quantity
                            </span>
                          </th>
                          <th
                            scope="col"
                          >
                            <span
                              class="text-uppercase p-2 text-black-50 text-nowrap"
                            >
                              Line Total
                            </span>
                          </th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr>
                          <th
                            scope="row"
                          >
                            <input
                              class="form-control"
                              data-testid="lineItems[0].item"
                              name="lineItems[0].item"
                              placeholder="Item"
                              type="text"
                            />
                          </th>
                          <td>
                            <input
                              class="form-control"
                              data-testid="lineItems[0].description"
                              name="lineItems[0].description"
                              placeholder="Description"
                              type="text"
                            />
                          </td>
                          <td>
                            <input
                              class="form-control"
                              data-testid="lineItems[0].unitCost"
                              name="lineItems[0].unitCost"
                              placeholder="Unit Cost"
                              type="number"
                            />
                          </td>
                          <td>
                            <input
                              class="form-control"
                              data-testid="lineItems[0].quantity"
                              name="lineItems[0].quantity"
                              placeholder="Quantity"
                              type="number"
                            />
                          </td>
                          <td
                            class="text-right"
                          >
                            <p
                              class="mb-0 mt-4"
                              data-testid="lineTotal"
                            >
                              £0.00
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="col-12 col-md-4 col-auto ml-auto p-0"
                    >
                      <div
                        class="row mt-5 pb-3"
                      >
                        <div
                          class="col-6 font-weight-bold"
                        >
                          Subtotal
                        </div>
                        <div
                          class="col-6 text-right"
                        >
                          <span
                            data-testid="subTotal"
                          >
                            £
                            0.00
                          </span>
                        </div>
                      </div>
                      <div
                        class="row pb-3 pt-3"
                      >
                        <div
                          class="col-6 font-weight-bold"
                        >
                          Tax
                        </div>
                        <div
                          class="col-6 text-right"
                        >
                          <span
                            data-testid="tax"
                          >
                            £
                            0.00
                          </span>
                        </div>
                      </div>
                      <div
                        class="row pb-3 pt-3"
                      >
                        <div
                          class="col-6 font-weight-bold"
                        >
                          Balance Due
                        </div>
                        <div
                          class="col-6 text-right"
                        >
                          <span
                            data-testid="balanceDue"
                          >
                            £
                            0.00
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="row mt-5"
                  >
                    <div
                      class="col-12"
                    >
                      <div
                        class="d-flex justify-content-center"
                      >
                        <button
                          class="btn btn-primary mr-3"
                          type="button"
                        >
                          Save
                        </button>
                        <button
                          class="btn btn-primary-soft mr-3"
                          type="button"
                        >
                          Save As Draft
                        </button>
                        <button
                          class="btn btn-primary-soft mr-3"
                          type="button"
                        >
                          Nag Client To Pay
                        </button>
                        <button
                          class="btn btn-primary-soft"
                          type="button"
                        >
                          Download PDF
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
